File: repos/appTiendaUPN/appTiendaUPN/Models/CarritoItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace appTiendaUPN.Models
{
    [Table("carritoitems")]
    public class CarritoItem
    {
        [Key]
        [Column("carritoitemid")]
        public int CarritoItemId { get; set; }

        [Required]
        [Column("carritoid")]
        public int CarritoId { get; set; }

        [Required]
        [Column("productoid")]
        public int ProductoId { get; set; }

        [Required]
        [Column("cantidad")]
        public int Cantidad { get; set; }

        [Required]
        [Column("precio", TypeName = "decimal(18,2)")]
        public decimal Precio { get; set; }

        // Relaciones
        [ForeignKey("CarritoId")]
        public Carrito? Carrito { get; set; }

        [ForeignKey("ProductoId")]
        public Producto? Producto { get; set; }
    }
}

File: repos/Soluciones/appFinal_Backend/src/main/java/com/example/appfinal_backend/model/CarritoItem.java
package com.example.appfinal_backend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "carritoitems")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CarritoItem {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "carritoitemid")
    private Integer carritoItemId;
    
    @Column(name = "carritoid", nullable = false)
    private Integer carritoId;
    
    @Column(name = "productoid", nullable = false)
    private Integer productoId;
    
    @Column(name = "cantidad", nullable = false)
    private Integer cantidad = 1;
    
    @Column(name = "precio", nullable = false, precision = 18, scale = 2)
    private BigDecimal precio;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "carritoid", insertable = false, updatable = false)
    private Carrito carrito;
    
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "productoid", insertable = false, updatable = false)
    private Producto producto;
}
